avro-fastserde/src/main/java/com/linkedin/avro/fastserde/FastSerdeCache.java

```diff
@@ -44,6 +44,22 @@
 
   private static volatile FastSerdeCache _INSTANCE;
 
+  /*
+   * Fast-avro will generate and load serializer and deserializer(SerDes) classes into metaspace during runtime.
+   * During serialization and deserialization, fast-avro also leverages JIT compilation to boost the SerDes speed.
+   * And JIT compilation code is saved in code cache.
+   * Too much usage of metaspace and code cache will bring GC/OOM issue.
+   *
+   * We set a hard limit of the total number of SerDes classes generated and loaded by fast-avro.
+   * Currently we set the limit to 400 based on our performance test with manually generated schemas.
+   * The number could keep the metaspace and codecache usage within a safe range.
+   * Fast-avro will fall back to regular avro after the limit is hit.
+   *
+   * TODO: Let users configure the limit.
```

Comment:
I don't think we should merge this in before making it configurable...

Reply:
Hi, I added set/get limit functions in the cache. And also added getFastSerdeCache function in datum reader/writer. So now it is configurable.